app/controllers/users/sessions_controller.rb

```diff
@@ -13,11 +13,17 @@
 class Users::SessionsController < Devise::SessionsController
   before_action :configure_sign_in_params, only: :create
 
-  prepend_before_action :authenticate_with_otp_two_factor,
+  prepend_before_action :handle_otp_flow,
     if: -> { action_name == "create" && otp_two_factor_enabled? }
```

Comment:
If we're looking to simplify things, can we just remove `otp_required_for_login` method and the `otp_required_for_login` field?

Reply:
It's part of the `two-factor` gem. I'm not certain it's worth the effort against defaulting the column to `true` in a `before_initialize` or similar, which at least gives us the option of turning it off to speed development.

Reply:
I understand that it seems foolish to disable something we get for free, but I really don't see us needing this feature (MFA is required for all users of the system), and I'm keen to remove as much logic as we can from an already overly confused process.

Reply:
It's in use [here](https://github.com/tinfoil/devise-two-factor/blob/main/lib/devise_two_factor/strategies/two_factor_authenticatable.rb#L23)

Reply:
We've agreed to put [this card](https://trello.com/c/1b4Z4IJF/2432-remove-the-usersotprequiredforlogin-column) in the Tech debt column.